faunadb-java-dsl/src/main/java/com/faunadb/client/types/Types.java

```diff
@@ -0,0 +1,233 @@
+    KNOWN_TYPES.put(Integer.class, new SimpleType(Integer.class));
+    KNOWN_TYPES.put(Short.class, new SimpleType(Short.class));
+    KNOWN_TYPES.put(Byte.class, new SimpleType(Byte.class));
+    KNOWN_TYPES.put(Character.class, new SimpleType(Character.class));
+    KNOWN_TYPES.put(Float.class, new SimpleType(Float.class));
+    KNOWN_TYPES.put(Double.class, new SimpleType(Double.class));
+  }
+
+  /**
+   * <p>Creates a generic collection type representation</p>
+   * <p>
+   * <pre>{@code
+   * Types.collectionOf(LinkedList.class, User.class)
+   * }</pre>
+   *
+   * @param collectionType A concrete collection type
+   * @param elementType    An element type of the collection
+   * @return A type representing a collection
+   */
+  public static CollectionType collectionOf(Class<? extends Collection> collectionType, Type elementType) {
+    return new CollectionType(collectionType, of(elementType));
+  }
+
+  /**
+   * <p>Creates a collection type specialized for {@link ArrayList}</p>
+   * <p>
+   * <pre>{@code
+   * Types.arrayListOf(User.class)
+   * Types.arrayListOf(String.class)
+   * Types.arrayListOf(int.class)
+   * }</pre>
+   *
+   * @param elementType An element type of the collection
+   * @return A type representing an {@link ArrayList}
+   */
+  public static CollectionType arrayListOf(Type elementType) {
+    return collectionOf(ArrayList.class, elementType);
+  }
+
+  /**
+   * <p>Creates a collection type specialized for {@link HashSet}</p>
+   * <p>
+   * <pre>{@code
+   * Types.hashSetOf(User.class)
+   * Types.hashSetOf(String.class)
+   * Types.hashSetOf(int.class)
+   * }</pre>
+   *
+   * @param elementType An element type of the collection
+   * @return A type representing a {@link HashSet}
+   */
+  public static CollectionType hashSetOf(Type elementType) {
+    return collectionOf(HashSet.class, elementType);
+  }
+
+  /**
+   * <p>Creates a generic map type representation</p>
+   * <p>
+   * <pre>{@code
+   * Types.mapOf(Hashtable.class, String.class, User.class)
+   * }</pre>
+   *
+   * @param mapType   A concrete map type
+   * @param keyType   The type of key in the map
+   * @param valueType The type of the value in the map
+   * @return A type representing a map
+   */
+  public static MapType mapOf(Class<? extends Map> mapType, Type keyType, Type valueType) {
+    if (keyType != String.class && keyType != STRING_TYPE)
+      throw new FaunaException("Only string keys are supported at the moment for maps");
+
+    return new MapType(mapType, of(keyType), of(valueType));
+  }
+
+  /**
+   * <p>Creates a map type specialized for {@link HashMap}</p>
+   * <p>
+   * <pre>{@code
+   * Types.hashMapOf(String.class, User.class)
+   * }</pre>
+   *
+   * @param keyType   The type of key in the map
+   * @param valueType The type of the value in the map
+   * @return A type representing a {@link HashMap}
+   */
+  public static MapType hashMapOf(Type keyType, Type valueType) {
```

Comment:
I think this also needs to enforce string only keys. Otherwise it can be misleading.